scripts/zkai_attention_kv_d16_two_head_longseq_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,1060 @@
+    proof_bytes = json.dumps(payload, separators=(",", ":"), ensure_ascii=False).encode("utf-8")
+    if len(proof_bytes) != len(envelope["proof"]):
+        raise AttentionKvD16TwoHeadLongseqFusedSoftmaxTableGateError("same-size proof mutation changed byte length")
+    envelope["proof"] = list(proof_bytes)
+
+
+def assert_fields(mapping: dict[str, Any], expected: dict[str, Any], label: str) -> None:
+    for key, expected_value in expected.items():
+        if mapping.get(key) != expected_value:
+            raise AttentionKvD16TwoHeadLongseqFusedSoftmaxTableGateError(
+                f"{label} drift for {key}: got {mapping.get(key)!r}"
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)